src/Actions/Actions/SelectAction.cs

```diff
@@ -85,6 +84,7 @@ private set
                 if (_POIElementContext != null)
                 {
                     dma.RemoveElementContext(_POIElementContext.Id);
+                    _POIElementContext.Dispose();
```

Comment:
>_POIElementContext.Dispose(); [](start = 20, length = 29)

This is a behavioral change. Previous code would leave the previous ElementContext undisposed. Any concerns from this?

Reply:
Is the property set to null when the object is destroyed? In other words, what guarantees the properties set method will be called with a null value?

Reply:
The current Dispose method both Disposes and sets to null, bringing up the potential for a double disposal. I checked the `ElementContext` class, and double disposal isn't a problem there. I did a quick audit of our calling of Dispose and was troubled by the various inconsistencies. I've made a mental note for a future pass to try to be more consistent, but for now I think we're ok

---
In reply to: [282939931](https://github.com/microsoft/axe-windows/pull/23#discussion_r282939931) [](ancestors = 282939931)